### PR TITLE
Convert additional Editing classes to kotlin.

### PR DESCRIPTION
**Pull request by dbrant**

### CODE REVIEW

File: app/src/main/java/org/wikipedia/edit/preview/EditPreviewFragment.kt [hunk 1]
```diff
@@ -0,0 +1,301 @@
...
+    override val toolbarMargin = 0
+
+    override val referencesGroup: List<PageReferences.Reference>?
+        get() = references.referencesGroup
+
+    override val selectedReferenceIndex: Int
+        get() = references.selectedIndex
+
+    val isActive: Boolean
+        get() = binding.editPreviewContainer.visibility == View.VISIBLE
```
Comment (cooltey) [inline app/src/main/java/org/wikipedia/edit/preview/EditPreviewFragment.kt:67]: Explicit type specification can be reduced.

File: app/src/main/java/org/wikipedia/edit/preview/EditPreviewFragment.kt [hunk 2]
```diff
@@ -0,0 +1,301 @@
...
+            R.string.edit_summary_tag_links
+        )
+        val strings = L10nUtil.getStringsForArticleLanguage(pageTitle, summaryTagStrings)
+
+        summaryTags.clear()
+        for (i in summaryTagStrings) {
+            val tag = EditSummaryTag(activity)
+            tag.text = strings[i]
+            tag.tag = i
+            tag.setOnClickListener { view: View ->
```
Comment (cooltey) [inline app/src/main/java/org/wikipedia/edit/preview/EditPreviewFragment.kt:119]: Explicit type specification can be removed.

File: app/src/main/java/org/wikipedia/edit/preview/EditPreviewFragment.kt [hunk 3]
```diff
@@ -0,0 +1,301 @@
...
+                funnel.logEditSummaryTap((view.tag as Int))
```
Comment (cooltey) [inline app/src/main/java/org/wikipedia/edit/preview/EditPreviewFragment.kt:120]: Redundant parentheses can be removed.

File: app/src/main/java/org/wikipedia/edit/preview/EditPreviewFragment.kt [hunk 4]
```diff
@@ -0,0 +1,301 @@
...
+                if (!isAdded) {
+                    return
+                }
+                (requireActivity() as EditSectionActivity).showProgressBar(false)
+                requireActivity().invalidateOptionsMenu()
+                bridge.execute(JavaScriptActionHandler.setTopMargin(0))
+            }
+        }
+
+        bridge.addListener("setup", object : JSEventListener {
```
Comment (cooltey) [inline app/src/main/java/org/wikipedia/edit/preview/EditPreviewFragment.kt:196]: The `interface JSEventListener` can be `fun interface JSEventListener`.

File: app/src/main/java/org/wikipedia/edit/preview/EditPreviewFragment.kt [hunk 5]
```diff
@@ -0,0 +1,301 @@
...
+         * Shows the user a dialogue asking them if they really meant to leave the edit
+         * workflow, and warning them that their changes have not yet been saved.
+         *
+         * @param runnable The runnable that is run if the user chooses to leave.
+         */
+        private fun showLeavingEditDialogue(runnable: Runnable) {
+            // Ask the user if they really meant to leave the edit workflow
+            val leavingEditDialog = AlertDialog.Builder(requireActivity())
+                .setMessage(R.string.dialog_message_leaving_edit)
+                .setPositiveButton(R.string.dialog_message_leaving_edit_leave) { dialog: DialogInterface, _: Int ->
```
Comment (cooltey) [inline app/src/main/java/org/wikipedia/edit/preview/EditPreviewFragment.kt:286]: Type specification can be removed.

File: app/src/main/java/org/wikipedia/edit/summaries/EditSummaryFragment.kt [hunk 6]
```diff
@@ -0,0 +1,112 @@
...
+
+    /**
+     * Gets the custom ("other") summary, if any, that the user has entered.
+     * @return Custom summary of the edit.
+     */
+    val summary: String
+        get() = binding.editSummaryEdit.text.toString()
+
+    val isActive: Boolean
+        get() = binding.root.visibility == View.VISIBLE
```
Comment (cooltey) [inline app/src/main/java/org/wikipedia/edit/summaries/EditSummaryFragment.kt:31]: Type specification can be removed.

File: app/src/main/java/org/wikipedia/edit/summaries/EditSummaryFragment.kt [hunk 7]
```diff
@@ -0,0 +1,112 @@
...
+import org.wikipedia.edit.EditSectionActivity
+import org.wikipedia.page.PageTitle
+import org.wikipedia.util.DeviceUtil
+import org.wikipedia.views.ViewAnimations
+
+class EditSummaryFragment : Fragment() {
+    lateinit var title: PageTitle
+    private lateinit var editSummaryHandler: EditSummaryHandler
+    private var _binding: FragmentPreviewSummaryBinding? = null
+    private val binding get() = _binding!!
```
Comment (cooltey) [inline app/src/main/java/org/wikipedia/edit/summaries/EditSummaryFragment.kt:21]: Would it be possible to move these to the top?

File: app/src/main/java/org/wikipedia/edit/summaries/EditSummaryFragment.kt [hunk 8]
```diff
@@ -0,0 +1,112 @@
...
+    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View {
+        _binding = FragmentPreviewSummaryBinding.inflate(layoutInflater, container, false)
+
+        // Explicitly enable standard dictionary autocompletion in the edit summary box
+        // We should be able to do this in the XML, but doing it there doesn't work. Thanks Android!
+        binding.editSummaryEdit.inputType = binding.editSummaryEdit.inputType and EditorInfo.TYPE_TEXT_FLAG_AUTO_COMPLETE.inv()
+
+        // ...so that clicking the "Done" button on the keyboard will have the effect of
+        // clicking the "Next" button in the actionbar:
+        binding.editSummaryEdit.setOnEditorActionListener { _: TextView?, actionId: Int, keyEvent: KeyEvent? ->
```
Comment (cooltey) [inline app/src/main/java/org/wikipedia/edit/summaries/EditSummaryFragment.kt:42]: Type specification can be removed.

**Review by cooltey — CHANGES_REQUESTED** [review]

File: app/src/main/java/org/wikipedia/edit/preview/EditPreviewFragment.kt (hunk 4 above)
Comment (dbrant) [reply app/src/main/java/org/wikipedia/edit/preview/EditPreviewFragment.kt:196]: correct!

**Pull request merged**